ktor-http/ktor-http-cio/jvm/src/io/ktor/http/cio/RemoteHost.kt

```diff
@@ -0,0 +1,27 @@
+package io.ktor.http.cio
+
+import io.ktor.network.sockets.*
+import java.net.*
+import java.nio.channels.*
+
+fun remoteHost(socket: AConnectedSocket) = remoteHost(socket.remoteAddress)
+
+fun remoteHost(socketChannel: SocketChannel) = remoteHost(socketChannel.remoteAddress)
+
+
+
+fun remoteHost(socketAddress: SocketAddress): CharSequence {
+    return (socketAddress as? InetSocketAddress)?.address?.let { remoteHost(it) } ?: ""
+}
+
+fun remoteHost(address: InetAddress) = when (address) {
+    is Inet4Address -> address.address.joinToString(".") {
+        (it.toInt() and 0xFF).toString()
+    }
+    is Inet6Address -> address.address.toList().chunked(2).joinToString(":") { pair ->
```

Comment:
Creating so many instances for every request may significantly slowdown server at high rates because of high memory pressure. Unfortunately, this needs to be rewritten in a more optimal manner. Btw, doesn't `Inet6Address` already have a proper formatting function? What about `Inet6Address.getHostAddress`?